test/boost/reader_concurrency_semaphore_test.cc

```diff
@@ -718,6 +718,12 @@ SEASTAR_THREAD_TEST_CASE(test_reader_concurrency_semaphore_admission) {
     reader_concurrency_semaphore semaphore(reader_concurrency_semaphore::for_tests{}, get_name(), initial_resources.count, initial_resources.memory);
     auto stop_sem = deferred_stop(semaphore);
 
+    auto timeout = [] () {
+        // The timeout shall exceed the one used in ::require_can_admit, otherwise some tests cases may fail
+        // due to timeouts and consequently invalid permit's state.
+        return db::timeout_clock::now() + 100ms;
+    };
+
```

Comment:
I'm worried that this will introduce flakiness.

Reply:
Actually, it turns out it does. The test failed 1/30 in the release mode. Last time it was in the debug mode.

Some change is required otherwise permits get aborted at admission and tests will fail. Probably, we can simply use `db::no_timeout`